### ISSUE TITLE
Vulkan: Alpha channel not included in Viewport capture due to `SCREEN_TEXTURE` and `DEPTH_TEXTURE` not being updated

### ISSUE THREAD

**Issue by supagu** [godotengine/godot#35230]
**Godot version:** Vulkan

**OS/device including version:** Linux/Arch/Manajro

**Issue description:** I have a ViewportContainer with a Viewport child to render something in the viewport. I have a custom ShaderMaterial applied to the ViewportContainer 
with some code like this:

```
shader_type canvas_item;

uniform float width: hint_range(0.0, 10.0);
//uniform vec4 color: hint_color;

void fragment() {
	vec4 c = texture(TEXTURE, SCREEN_UV, 0.0);
        COLOR=c.aaaa;
}
```
![image](https://user-images.githubusercontent.com/11153232/72604606-1c75fd80-396b-11ea-865c-30ca2193447e.png)

the texture always comes out white.

what I am rendering in the viewport is a mesh with a material that is like so:
![image](https://user-images.githubusercontent.com/11153232/72604441-d0c35400-396a-11ea-964a-d4ea8bd3e779.png)

so that it draws in the alpha channel.

this worked fine when in the godot 3.2 branch and seems to have broken when I try it in Vulkan branch,

**Comment by supagu** [comment]
Additionally, I tried setting the environment clear color to (0, 0, 0, 0) but still all I see is white in the alpha channel.

**Comment by supagu** [comment]
in fact any work I have tried to do with capture textures has not worked well.

**Comment by briansemrau** [comment]
This issue is still relevant as of 7a31baf3b9dc066fb15af8aa4f90193df3615226. After experimenting with spatial materials, it appears that `SCREEN_TEXTURE` and `DEPTH_TEXTURE` are not being updated with the new Vulkan renderer, resulting in texture calls returning a default value. @Calinou Maybe this issue should be renamed and possibly labeled as a regression?

**Comment by flakusha** [comment]
This issue still exists.
Shader which does work correctly in Godot 3.2.3 does not capture SCREEN_TEXTURE in Godot 4 dev 94e5f66cbb399ffd3e501a237d7eb3de79ce1607
![Godot_v3 2 3-stable_win64_DAu4KqV7LL](https://user-images.githubusercontent.com/14177728/96336926-9dddf600-108b-11eb-9504-e89b22b11bff.png)
![godot windows opt tools 64_XwdTv0YFPl](https://user-images.githubusercontent.com/14177728/96336944-bb12c480-108b-11eb-846a-0c62fea5934f.png)

**Comment by Calinou** [comment]
This appears to be fixed as of 4.0.rc2, closing.

![image](https://user-images.githubusercontent.com/180032/220201725-69b6772d-3ecd-4645-a369-3a3d339bc5c1.png)

![image](https://user-images.githubusercontent.com/180032/220201764-479e2ea1-8f29-4395-84ee-c91c813697e6.png)

**Testing project:** [test_shader_viewport.zip](https://github.com/godotengine/godot/files/10787921/test_shader_viewport.zip)
(`UV` is used instead of `SCREEN_UV` for sampling, but it that's purely for cosmetic reasons.)
